components/org.wso2.carbon.identity.scim.common/src/main/java/org/wso2/carbon/identity/scim/common/utils/SCIMCommonUtils.java

```diff
@@ -217,6 +222,43 @@ public static void setAdminSCIMAttributes() {
         }
     }
 
+    public static void addAdminGroup(UserStoreManager userStoreManager) throws UserStoreException {
+        try {
+            SCIMGroupHandler scimGroupHandler = new SCIMGroupHandler(userStoreManager.getTenantId());
+
+            String domainName = UserCoreUtil.getDomainName(userStoreManager.getRealmConfiguration());
+            if (domainName == null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Domain name is null and setting default domain as " + UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME);
+                }
+                domainName = UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME;
+            }
+            String roleNameWithDomain = UserCoreUtil
+                    .addDomainToName(userStoreManager.getRealmConfiguration().getAdminRoleName(), domainName);
+            // UserCore Util functionality does not append primary
+            roleNameWithDomain = SCIMCommonUtils.getGroupNameWithDomain(roleNameWithDomain);
```

Comment:
Need to check SCIMCommonUtils.getGroupNameWithDomain() add primary domain correctly

Reply:
Fixed getGroupNameWithDomain method.